catalog/registry/catalog-registry-federationadminserviceimpl/src/main/java/ddf/catalog/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -0,0 +1,305 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.Marshaller;
+import javax.xml.datatype.DatatypeConstants;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.configuration.SystemInfo;
+import org.codice.ddf.parser.Parser;
+import org.codice.ddf.parser.ParserConfigurator;
+import org.codice.ddf.parser.ParserException;
+import org.codice.ddf.security.common.Security;
+import org.geotools.filter.FilterFactoryImpl;
+import org.opengis.filter.Filter;
+import org.opengis.filter.FilterFactory;
+import org.opengis.filter.sort.SortBy;
+import org.opengis.filter.sort.SortOrder;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.registry.api.metacard.RegistryObjectMetacardType;
+import ddf.catalog.registry.common.RegistryConstants;
+import ddf.catalog.registry.federationadmin.service.FederationAdminService;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ExtrinsicObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.InternationalStringType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.LocalizedStringType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ObjectFactory;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ValueListType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.VersionInfoType;
+
+public class FederationAdminServiceImpl implements FederationAdminService {
+
+    private static final XLogger LOGGER = new XLogger(LoggerFactory.getLogger(
+            FederationAdminServiceImpl.class));
+
+    private static final FilterFactory FILTER_FACTORY = new FilterFactoryImpl();
+
+    private static final ObjectFactory RIM_FACTORY = new ObjectFactory();
+
+    private static final String REGISTRY_CONTENT_TYPE = "registry.service";
+
+    private static final String REGISTRY_NODE_PRIMARY = "PRIMARY";
```

Comment:
Get rid of this once you change the other thing we talked about before to a boolean.